Uml.Robotics.Ros/RemappingHelper.cs

```diff
@@ -41,13 +42,23 @@ public static bool GetRemappings(ref string[] args, out IDictionary<string, stri
             // If ROS.ROS_MASTER_URI was not explicitely set by the program calling Init, and was not passed in as a remapping argument, then try to find it in ENV.
             if (string.IsNullOrEmpty(ROS.ROS_MASTER_URI))
             {
-                ROS.ROS_MASTER_URI = System.Environment.GetEnvironmentVariable("ROS_MASTER_URI");
+                IDictionary _vars;
+
+                //check user env first, then machine if user doesn't have uri defined.
+                if ((_vars = Environment.GetEnvironmentVariables(EnvironmentVariableTarget.User)).Contains("ROS_MASTER_URI")
+                    || (_vars = Environment.GetEnvironmentVariables(EnvironmentVariableTarget.Machine)).Contains("ROS_MASTER_URI"))
+                    ROS.ROS_MASTER_URI = (string)_vars["ROS_MASTER_URI"];
```

Comment:
Why not using the process environment variables? They already should contain machine + user variables...

Reply:
I had a series of problems reading these variables, I solved by making the reading explicit